### PR TITLE
docs: job queue write feature

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
Add documentation for the full job queue.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/features/job-queue.txt [hunk 1]
```diff
@@ -4,63 +4,136 @@
...
+************
+ Job States
+************
+
+Jobs in the Job Queue can be in the ``Queued`` or ``Scheduled`` states. A ``Queued`` job has been
+received by Determined but has not been allocated resources yet. Any change made to queued jobs
+should take effect as soon as they have been received by Determined. A job in the ``Scheduled``
+state has already been chosen by the scheduler to start execution and it may have already acquired
+resources or be running at that moment. Once it has started running, a progress bar like the one
+seen in the experiments section should appear. A job that has already started to run will not be
```
Comment (hamidzr) [inline docs/features/job-queue.txt:27]: clarify that this is only the case if the job supports reporting progress. (so the users don't expect this for other job types)

File: docs/features/job-queue.txt [hunk 2]
```diff
@@ -4,63 +4,136 @@
...
-Jobs can be in the Queued or Scheduled states. Queued jobs have been received by Determined but have
-not been allocated resources yet. Any change made to those experiments in the Job Queue should take
-effect as soon as they are received. A job in the Scheduled state has selected to run by the
-scheduler; it may or may not have acquired resources or be running at that moment. Once it has
-started running, a progress bar like the one in the experiments tab should appear. A job that has
-already started to run will not reflect changes immediately and will instead need to be preempted.
+   $ det job list --resource-pool compute
 
-Jobs that have completed or errored out are not counted as active and are omitted from this list.
+For more information about the CLI options, see the CLI documentation or use ``det job list -h``.
```
Comment (hamidzr) [inline docs/features/job-queue.txt:51]: 👍🏼 I was just about to ask for this.

File: docs/features/job-queue.txt [hunk 3]
```diff
@@ -4,63 +4,136 @@
...
+
+Both the WebUI and the CLI will display a table of results, ordered by scheduling order. Note that
```
Comment (hamidzr) [inline docs/features/job-queue.txt:53]: something like
.. by the scheduling order _if available_ (eg supported by the scheduler type)

**Review by hamidzr — APPROVED** [review]
thanks for updating the docs. I had 2 comments but otherwise lgtm

**Review by gh-determined-ai — CHANGES_REQUESTED** [review]
See https://hpe-my.sharepoint.com/:w:/p/gene_hartsell/EaLrSVnLWH9DvqmI8gMzuagB2iNwemRC7GPMlmppGpXoiQ?e=ntvZbz

**Comment by hamidzr** [comment]
> https://hpe-my.sharepoint.com/:w:/p/gene_hartsell/EaLrSVnLWH9DvqmI8gMzuagB2iNwemRC7GPMlmppGpXoiQ?e=ntvZbz

I'm helping @eecsliu to wrap this up. I'm not sure what the ideal approach is to address the comments you had on the shared doc. I'm thinking I'll go through and apply the comments and introduce my own changes. then you can take another look after?

**Comment by hamidzr** [comment]
note: we need to take out k8 mentions in the queue modification section

**Comment by gh-determined-ai** [comment]
@hamidzr  The "official" right approach is here: https://determinedai.atlassian.net/wiki/spaces/DOC/pages/1101594663. But nobody's successfully followed that yet. Do what you can or want to with my comments and I'll follow with a quick review. All of the content needs to eventually get this treatment so you can either do it now or it will be taken care of along with everything else, later. Thanks.

File: docs/features/job-queue.txt (hunk 1 above)
Comment (hamidzr) [reply docs/features/job-queue.txt:27]: I addressed this in my patch

**Comment by hamidzr** [comment]
I molded the document back to rst format so we're closer to something releasable. I need to see what changed but I don't seem to get html output from `make -C docs build` to check out the final output. 

[jqwrite-docs.patch.txt](https://github.com/determined-ai/determined/files/8328356/jqwrite-docs.patch.txt)

File: docs/features/job-queue.txt [hunk 4]
```diff
@@ -4,63 +4,135 @@
...
+
+.. code::
+
+   $ det job list --resource-pool compute-pool
+
+For more information about the CLI options, see the CLI documentation or use the ``det job list -h``
+command.
+
+The WebUI and the CLI display a table of results, ordered by scheduling order. The scheduling order
+does not represent the job priority. In addition to job order, the table incudes the job states and
```
Comment (gh-determined-ai) [inline docs/features/job-queue.txt:53]: Spelling: "incudes"

File: docs/features/job-queue.txt [hunk 5]
```diff
@@ -4,63 +4,135 @@
...
+=====
 
-Users may modify the job queue in either the WebUI or the CLI.
+To modify the job queue in the Webui,
+
+#. Go to the **Job Queue** section.
+#. Find the job to modify.
+#. Click the three dots in the right-most column of the job.
+#. Find and click the **Manage Job** option.
+#. Make the change you want on the pop up page, and click **OK**.
```
Comment (gh-determined-ai) [inline docs/features/job-queue.txt:78]: hyphenate: "pop-up"

File: docs/features/job-queue.txt [hunk 6]
```diff
@@ -4,63 +4,135 @@
...
 ***********************
- Job Queue Limitations
+ Job Queue Constraints
 ***********************
 
-.. code:: markdown
-
-   TODO
-   -  define and describe different states and their meanings
+Using the job queue to modify jobs has a number of constraints:
```
Comment (gh-determined-ai) [inline docs/features/job-queue.txt:128]: I think "There are a number of constraints associated with using the job queue to modify jobs:" might sound a little better.

File: docs/features/job-queue.txt [hunk 7]
```diff
@@ -4,63 +4,135 @@
...
+ Modify the Job Queue
+**********************
 
-Jobs can be in the Queued or Scheduled states. Queued jobs have been received by Determined but have
-not been allocated resources yet. Any change made to those experiments in the Job Queue should take
-effect as soon as they are received. A job in the Scheduled state has selected to run by the
-scheduler; it may or may not have acquired resources or be running at that moment. Once it has
-started running, a progress bar like the one in the experiments tab should appear. A job that has
-already started to run will not reflect changes immediately and will instead need to be preempted.
+The job queue can be changed in the WebUI **Job Queue** Section or by using the CLI ``det job
```
Comment (gh-determined-ai) [inline docs/features/job-queue.txt:60]: "... Section ..." should be lowercase: "... section ..."

**Review by gh-determined-ai — APPROVED** [review]

**Pull request merged**